file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -303,7 +345,75 @@ protected internal override async Task OnExecuteCommand()
                 IsCommitting = false;
                 IsExecuting = false;
                 IsExecutingComplete = true;
+                IsCanceled = false;
+                IsCommittingOnCancel = false;
+                batchHandle = null;
+                CancellationTokenSource.Dispose();
+            }
+        }        
+
+        private void OnCancelJobCommand()
+        {            
+            IsCanceled = false;            
+            logger.LogInformation("Cancel job requested for Action : {Action}, displayName:{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+            string msg = batchHandle !=null ? string.Format("Are you sure you want to cancel this job batch ID: {0}?", batchHandle.BatchId): "Are you sure you want to cancel this job?";            
+            MessageBoxResult result = MessageBox.Show(msg, "Cancel Job Confirmation", MessageBoxButton.YesNo, MessageBoxImage.Question);
+            if (result == MessageBoxResult.Yes)
+            {               
+                if (CancellationTokenSource != null && !CancellationTokenSource.IsCancellationRequested)
+                {
+                    logger.LogInformation("Cancel job confirmed for Action : {Action}, displayName:{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+                    IsCanceled = true;
+                    IsCommitting = false;                                                               
+                    CancellationTokenSource.Cancel();                                        
+                } 
+            }
+            else
+            {
+                logger.LogInformation("Cancel job canceled for Action : {Action} and displayName :{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+            }
+        }
+
+        private async Task<string> HandleCanceledOperationsAsync(IBatchHandle batchHandle, bool IsCommitting, IFileShareApiAdminClient fileShareClient, OperationCanceledException e, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested && e.CancellationToken == cancellationToken)
```

Comment:
Method `HandleCanceledOperationsAsync` is private and called from only one place. So do we really need this check and parameters `cancellationToken` & `e`?

Reply:
Thank you Vijendra for reviewing changes. Agree with your review comment as now method is called from only one place so removed this check and parameters cancellationToken & e from method HandleCanceledOperationsAsync.